handbook/engineering/web/frontend-platform/wildcard-component-library.md

```diff
@@ -0,0 +1,66 @@
+# Wildcard Component Library (WCL)
+
+Wildcard Component Library is a comprehensive collection of reusable, design-approved components that can be composed to quickly build out different user journeys across the web app. WCL is stored in its own [directory](https://github.com/sourcegraph/sourcegraph/tree/main/client/wildcard), owned by the [Fronted Platform Team](https://about.sourcegraph.com/handbook/engineering/web/frontend-platform) and contributed to by all Sourcegraph teammates. WCL is a part of the Wildcard Design System.
+
+## WCL x Frontend Platfrom Team
+- The Frontend Platform Team is the owner of the design system and Wildcard Component Library  and needs to be up-to-date with all the changes and needs of other teams
+- The Frontend Platform Team should enable other teams to make changes to the Wildcard Component Library by keeping them up-to-date (see Sharing updates to the components system), helping them plan their work, providing expertise, and reviewing the changes
+- The Frontend Platform team **shouldn't be responsible for implementing components for other teams**. Otherwise, we would create a bottleneck and FE Platform team wouldn’t have time to focus on their team goals.
+- Other product teams should feel empowered to make changes and create components by themselves
+
+## Changing and creating components
+### Process of changing or creating a component by a product team
```

Comment:
I can't remember if I've discussed this before. It feels a bit off that we have two separate processes depending on where the change originates from. The processes are quite similar, and I feel it opens up a grey area for work that is a collaboration across teams.

### Process of changing or creating a component

1. The designer creates a Figma file with full specification regarding the component (separate from the general project file or general design system files)
2. The designer initiates the cross-team communication by marking as reviewers:
  - Members of the engineering team who will be implementing the work
  - At least one other Design Team member
  - Members of other teams affected by the changes 
3. After all the reviews are completed, the engineering team plans the implementation
4. During the implementation, the engineering team stays in constant communication with the Design Team members engaged in the project. Asks for review, if necessary.
5. The engineering team should ask for final reviews from:
  - Members of the FE Platform team
  - Members of the engineering team implementing the change
  - Members of other teams affected by the changes (if applicable)
  - Members of the Design Team engaged in the project
7. When changes are approved, the engineering team merges the PR

Reply:
As discussed at the Frontend Crew meeting, we also want to create a Github issue for a new component to track the implementation progress.